openeo/extra/job_management.py

```diff
@@ -886,3 +936,191 @@ def create_job_db(path: Union[str, Path], df: pd.DataFrame, *, on_exists: str =
     else:
         raise NotImplementedError(f"Initialization of {type(job_db)} is not supported.")
     return job_db
+
+
+class UDPJobFactory:
+    """
+    Batch job factory based on a parameterized process definition
```

Comment:
I wonder whether UDP Job Factory is the clearest name. Without having read the function definition, I am not really sure what to expect. The definition does not necessarily clarify it either. 

Does factory here mean that we create jobs based on UDPs?

Reply:
I already decided to pick ProcessBasedJobCreator, which should already improve the situation.

but still open for better options

Reply:
that option sounds good to me. 

Process in itself is also still a bit vague, but preferable over openeoprocessjobcreator